src/thenable.ts

```diff
@@ -0,0 +1,5 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+interface Thenable<T> extends PromiseLike<T> { }
```

Comment:
without this definition (which I took [here](https://github.com/Microsoft/vscode-languageserver-node/blob/master/jsonrpc/src/thenable.ts) )

```
C:\projects\langserver-typescript>.\node_modules\.bin\tsc -p .
node_modules/vscode-jsonrpc/lib/main.d.ts(8,67): error TS2304: Cannot find name 'Thenable'.
node_modules/vscode-jsonrpc/lib/main.d.ts(46,93): error TS2304: Cannot find name 'Thenable'.
node_modules/vscode-languageserver/lib/files.d.ts(2,83): error TS2304: Cannot find name 'Thenable'.
node_modules/vscode-languageserver/lib/files.d.ts(3,138): error TS2304: Cannot find name 'Thenable'.
node_modules/vscode-languageserver/lib/files.d.ts(5,158): error TS2304: Cannot find name 'Thenable'.
node_modules/vscode-languageserver/lib/files.d.ts(10,155): error TS2304: Cannot find name 'Thenable'.
node_modules/vscode-languageserver/lib/main.d.ts(146,86): error TS2304: Cannot find name 'Thenable'.
node_modules/vscode-languageserver/lib/main.d.ts(153,88): error TS2304: Cannot find name 'Thenable'.
node_modules/vscode-languageserver/lib/main.d.ts(160,92): error TS2304: Cannot find name 'Thenable'.
node_modules/vscode-languageserver/lib/main.d.ts(219,67): error TS2304: Cannot find name 'Thenable'.
src/master-connection.ts(80,10): error TS2322: Type '{}' is not assignable to type 'Definition'.
  Type '{}' is not assignable to type 'Location[]'.
    Property 'includes' is missing in type '{}'.
src/master-connection.ts(87,10): error TS2322: Type '{}' is not assignable to type 'Hover'.
  Property 'contents' is missing in type '{}'.
src/master-connection.ts(107,43): error TS2304: Cannot find name 'Thenable'.
src/master-connection.ts(111,12): error TS7006: Parameter 'result' implicitly has an 'any' type.
src/master-connection.ts(117,8): error TS7006: Parameter 'err' implicitly has an 'any' type.
```